chia/wallet/trade_manager.py

```diff
@@ -178,516 +136,338 @@ async def get_locked_coins(self, wallet_id: int = None) -> Dict[bytes32, WalletC
-                else:
-                    spend_bundle = SpendBundle.aggregate([spend_bundle, new_spend_bundle])
-
-            if spend_bundle is None:
-                return False, None, None
-
-            now = uint64(int(time.time()))
-            trade_offer: TradeRecord = TradeRecord(
-                confirmed_at_index=uint32(0),
-                accepted_at_time=None,
-                created_at_time=now,
-                my_offer=True,
-                sent=uint32(0),
-                spend_bundle=spend_bundle,
-                tx_spend_bundle=None,
-                additions=spend_bundle.additions(),
-                removals=spend_bundle.removals(),
-                trade_id=std_hash(spend_bundle.name() + bytes(now)),
-                status=uint32(TradeStatus.PENDING_ACCEPT.value),
-                sent_to=[],
+            all_coins: List[Coin] = [c for coins in coins_to_offer.values() for c in coins]
+            notarized_payments: Dict[Optional[bytes32], List[NotarizedPayment]] = Offer.notarize_payments(
+                requested_payments, all_coins
             )
-            return True, trade_offer, None
+            announcements_to_assert = Offer.calculate_announcements(notarized_payments)
+
+            all_transactions: List[TransactionRecord] = []
+            fee_left_to_pay: uint64 = fee
+            for wallet_id, selected_coins in coins_to_offer.items():
+                wallet = self.wallet_state_manager.wallets[wallet_id]
+                # This should probably not switch on whether or not we're spending a CAT but it has to for now
+
+                if wallet.type() == WalletType.COLOURED_COIN:
+                    txs = await wallet.generate_signed_transaction(
+                        [abs(offer_dict[int(wallet_id)])],
+                        [Offer.ph()],
+                        fee=fee_left_to_pay,
+                        coins=set(selected_coins),
+                        puzzle_announcements_to_consume=announcements_to_assert,
+                    )
+                    all_transactions.extend(txs)
+                else:
+                    tx = await wallet.generate_signed_transaction(
+                        abs(offer_dict[int(wallet_id)]),
+                        Offer.ph(),
+                        fee=fee_left_to_pay,
+                        coins=set(selected_coins),
+                        puzzle_announcements_to_consume=announcements_to_assert,
+                    )
+                    all_transactions.append(tx)
+
+                fee_left_to_pay = uint64(0)
+
+            transaction_bundles: List[Optional[SpendBundle]] = [tx.spend_bundle for tx in all_transactions]
+            total_spend_bundle = SpendBundle.aggregate(list(filter(lambda b: b is not None, transaction_bundles)))
+            offer = Offer(notarized_payments, total_spend_bundle)
+            return True, offer, None
+
         except Exception as e:
+            raise e
```

Comment:
LGTM is flagging this as the code below is unreachable. Is re-raising the expected behavior?

Reply:
no it is not, that was for debugging :)